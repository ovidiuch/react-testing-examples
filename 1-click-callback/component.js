import React from 'react';

export function CustomButton({ onClick }) {
  return <button onClick={onClick}>Click me nao</button>;
}
