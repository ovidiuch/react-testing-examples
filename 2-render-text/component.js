import React from 'react';

export function HelloMessage({ name }) {
  return <span>Hello {name}</span>;
}
