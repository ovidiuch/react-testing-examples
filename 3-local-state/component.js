import React, { Component } from 'react';

export class StatefulCounter extends Component {
  state = {
    count: 0
  };

  increment = () => {
    this.setState({ count: this.state.count + 1 });
  };

  render() {
    return (
      <div>
        Clicked {this.state.count} times{' '}
        <button onClick={this.increment}>+1</button>
      </div>
    );
  }
}
