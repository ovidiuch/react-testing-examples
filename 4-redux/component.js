import React from 'react';
import { connect } from 'react-redux';

function Counter({ count, increment }) {
  return (
    <div>
      Clicked {count} times <button onClick={increment}>+1</button>
    </div>
  );
}

function mapStateToProps({ count }) {
  return { count };
}

const mapDispatchToProps = {
  increment: () => ({ type: 'INCREMENT' })
};

export const ReduxCounter = connect(mapStateToProps, mapDispatchToProps)(
  Counter
);
