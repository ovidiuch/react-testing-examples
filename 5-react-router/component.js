import React from 'react';
import { withRouter } from 'react-router';
import { Link } from 'react-router-dom';

function UserPage({ match: { params } }) {
  return (
    <div>
      User #{params.userId} <Link to="/users/6">Next user</Link>
    </div>
  );
}

export const UserWithRouter = withRouter(UserPage);
