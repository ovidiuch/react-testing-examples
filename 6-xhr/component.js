import React, { Component } from 'react';
import axios from 'axios';

export class ServerCounter extends Component {
  state = {
    isSyncing: true,
    count: 0
  };

  componentDidMount() {
    axios('/count').then(({ data: { count } }) => {
      this.setState({ isSyncing: false, count });
    });
  }

  increment = () => {
    this.setState({ isSyncing: true });

    axios.post('/count').then(({ data: { count } }) => {
      this.setState({ isSyncing: false, count });
    });
  };

  render() {
    const { isSyncing, count } = this.state;

    return isSyncing ? (
      <div>Syncing...</div>
    ) : (
      <div>
        Clicked {count} times <button onClick={this.increment}>+1</button>
      </div>
    );
  }
}
