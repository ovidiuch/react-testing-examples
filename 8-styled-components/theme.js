export const themeLight = {
  background: 'white',
  text: 'black'
};

export const themeDark = {
  background: 'black',
  text: 'white'
};
