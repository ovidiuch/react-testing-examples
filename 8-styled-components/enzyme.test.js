// highlight{11-13}
import React from 'react';
import { create as mount } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { themeLight, themeDark } from './theme';
import { HelloMessage } from './component';

// Hoist helper functions (but not vars) to reuse between test cases
const getWrapper = ({ theme }) =>
  mount(
    <ThemeProvider theme={theme}>
      <HelloMessage>Hello world!</HelloMessage>
    </ThemeProvider>
  );

it('renders "light" theme', () => {
  // Render new instance in every test to prevent leaking state
  const wrapper = getWrapper({ theme: themeLight });

  expect(wrapper.toJSON()).toMatchSnapshot();
});

it('renders "dark" theme', () => {
  // Render new instance in every test to prevent leaking state
  const wrapper = getWrapper({ theme: themeDark });

  expect(wrapper.toJSON()).toMatchSnapshot();
});
