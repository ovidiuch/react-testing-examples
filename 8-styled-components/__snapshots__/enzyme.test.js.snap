// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "dark" theme 1`] = `
<span
  className="component__HelloMessage-nxtc05-0 eGaOhj"
>
  Hello world!
</span>
`;

exports[`renders "light" theme 1`] = `
<span
  className="component__HelloMessage-nxtc05-0 doKCJn"
>
  Hello world!
</span>
`;
